compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -961,13 +961,22 @@ fn check_param_wf(tcx: TyCtxt<'_>, param: &hir::GenericParam<'_>) -> Result<(),
                         hir_ty.span,
                         "using raw pointers as const generic parameters is forbidden",
                     ),
-                    _ => tcx.dcx().struct_span_err(
-                        hir_ty.span,
-                        format!("`{}` is forbidden as the type of a const generic parameter", ty),
-                    ),
+                    _ => {
+                        // Avoid showing "{type error}" to users. See #118179.
+                        if ty.references_error() {
+                            return Ok(());
+                        }
```

Comment:
Actually, this should just return `Err(ErrorGuaranteed)` here, rather than `Ok(())`, I think.

You can use the alternative `error_reported` function which is like `references_error`, but returns a `Result<(), ErrorGuaranteed>` for ease of `?` operator:

```suggestion
                        ty.error_reported()?;
```

Reply:
That makes sense! I guess it kinda depends on if we're checking if the parameters are well-formed, or if it's all well-formed... while checking the const parameters. 

Tests don't seem to mind either way, and there are no docs for any of these methods, so I'll go with this. 😄

(by the way, is there any reason behind the lack of docs on most `rustc` modules/items? would well-researched docs be appreciated?)